instaloader/instaloader.py

```diff
@@ -736,10 +736,16 @@ def download_stories(self,
             self.context.log("Retrieving all visible stories...")
         else:
             userids = [p if isinstance(p, int) else p.userid for p in userids]
+            profile_count = len(userids)
 
-        for user_story in self.get_stories(userids):
+        for i, user_story in enumerate(self.get_stories(userids), start=1):
             name = user_story.owner_username
-            self.context.log("Retrieving stories from profile {}.".format(name))
+            if profile_count is not None:
```

Comment:
Hi! `profile_count` is only set if `userids` is passed to the function. If not, such as e.g. when calling instaloader with `:stories`, this line will yield an error:

```
Retrieving all visible stories...
Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/alex/instaloader/instaloader/__main__.py", line 488, in <module>
    main()
  File "/home/alex/instaloader/instaloader/__main__.py", line 466, in main
    _main(loader,
  File "/home/alex/instaloader/instaloader/__main__.py", line 175, in _main
    instaloader.download_stories(fast_update=fast_update, storyitem_filter=storyitem_filter)
  File "/home/alex/instaloader/instaloader/instaloader.py", line 80, in call
    return func(instaloader, *args, **kwargs)
  File "/home/alex/instaloader/instaloader/instaloader.py", line 797, in download_stories
    if profile_count is not None:
UnboundLocalError: local variable 'profile_count' referenced before assignment
```